ctapipe/tools/stage1.py

```diff
@@ -0,0 +1,767 @@
+    selected_gain_channel = Field(None, "Array [n_pix] of gain channel used")
+
+
+def tel_type_string_to_int(tel_type):
+    """
+    convert a telescope type string (str(TelescopeDescription)) into an integer that
+    can be stored.
+
+    Parameters
+    ----------
+    tel_type: str
+        telescope type string like "SST_ASTRI_CHEC"
+
+    Returns
+    -------
+    int:
+        hash value
+    """
+    return np.int32(
+        int(hashlib.sha1(tel_type.encode("utf8")).hexdigest(), 16) % (10 ** 8)
+    )
+
+
+class ImageQualityQuery(QualityQuery):
+    """ for configuring image-wise data checks """
+    pass
+
+
+def expand_tel_list(tel_list, max_tels, index_map):
+    """
+    un-pack var-length list of tel_ids into 
+    fixed-width bit pattern by tel_index
+
+    TODO: use index_map to index by tel_index rather than tel_id so this can be a
+    shorter array of bools.
+    """
+    pattern = np.zeros(max_tels).astype(bool)
+    pattern[tel_list] = 1
+    return pattern
+
+
+def create_tel_id_to_tel_index_transform(sub):
+    """
+    build a mapping of tel_id back to tel_index:
+    (note this should be part of SubarrayDescription)
+    """
+    idx = np.zeros(max(sub.tel_indices) + 1)
+    for key, val in sub.tel_indices.items():
+        idx[key] = val
+
+    # the final transform then needs the mapping and the number of telescopes
+    return partial(expand_tel_list, max_tels=len(sub.tel) + 1, index_map=idx)
+
+
+class Stage1ProcessorTool(Tool):
+    name = "ctapipe-stage1-process"
+    description = __doc__ + f" This currently writes {DL1_DATA_MODEL_VERSION} DL1 data"
+    examples = """
+    To process data with all default values:
+    > ctapipe-stage1-process --input events.simtel.gz --output events.dl1.h5 --progress
+    
+    Or use an external configuration file, where you can specify all options:
+    > ctapipe-stage1-process --config stage1_config.json --progress 
+    
+    The config file should be in JSON or python format (see traitlets docs). For an 
+    example, see ctapipe/examples/stage1_config.json in the main code repo. 
+    """
+
+    output_filename = Unicode(
+        help="DL1 output filename", default_value="events.dl1.h5"
+    ).tag(config=True)
+
+    write_images = Bool(
+        help="Store DL1/Event/Image data in output", default_value=False
+    ).tag(config=True)
+
+    write_parameters = Bool(
+        help="Compute and store image parameters", default_value=True
+    ).tag(config=True)
+
+    compression_level = Int(
+        help="compression level, 0=None, 9=maximum", default_value=5, min=0, max=9
+    ).tag(config=True)
+
+    split_datasets_by = CaselessStrEnum(
+        values=["tel_id", "tel_type"],
+        default_value="tel_id",
+        help="Splitting level for the parameters and images datasets",
+    ).tag(config=True)
+
+    compression_type = CaselessStrEnum(
+        values=["blosc:zstd", "zlib"],
+        help="compressor algorithm to use. ",
+        default_value="zlib",
```

Comment:
Why use the slower, less compressing method as default? zstd was supported by pytables for quite some time now and does not require extra dependencies.